frontend/src/metabase/components/Schedule/Schedule.tsx

```diff
@@ -115,159 +84,106 @@ export const Schedule = ({
-          .replace(/^\[/, "")
-          .replace(/\]$/, ""),
-        [verb, <SelectFrequency key="frequency" {...frequencyProps} />],
-      );
-    }
-  } else if (scheduleType === "daily") {
-    // e.g. "Send daily at 12:00pm"
-    return fillScheduleTemplate(
-      c(
-        "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is a time like '12:00pm'",
-      ).t`${"{0}"} ${"{1}"} at ${"{2}"}`,
-      [
-        verb,
-        <SelectFrequency key="frequency" {...frequencyProps} />,
-        <SelectTime key="time" {...timeProps} />,
-      ],
-    );
-  } else if (scheduleType === "weekly") {
-    // e.g. "Send weekly on Tuesday at 12:00pm"
-    return fillScheduleTemplate(
-      c(
-        "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is a day like 'Tuesday', {3} is a time like '12:00pm'",
-      ).t`${"{0}"} ${"{1}"} on ${"{2}"} at ${"{3}"}`,
-      [
-        verb,
-        <SelectFrequency key="frequency" {...frequencyProps} />,
-        <SelectWeekday key="weekday" {...weekdayProps} />,
-        <SelectTime key="time" {...timeProps} />,
-      ],
-    );
-  } else if (scheduleType === "monthly") {
-    // e.g. "Send monthly on the 15th at 12:00pm"
-    if (schedule.schedule_frame === "mid") {
-      return fillScheduleTemplate(
-        c(
-          "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is the noun '15th' (as in 'the 15th of the month'), {3} is a time like '12:00pm'",
-        ).t`${"{0}"} ${"{1}"} on the ${"{2}"} at ${"{3}"}`,
-        [
-          verb,
-          <SelectFrequency key="frequency" {...frequencyProps} />,
-          <SelectFrame key="frame" {...frameProps} />,
-          <SelectTime key="time" {...timeProps} />,
-        ],
-      );
-    } else {
-      // e.g. "Send monthly on the first Tuesday at 12:00pm"
-      return fillScheduleTemplate(
-        c(
-          "{0} is a verb like 'Send', {1} is an adverb like 'hourly', {2} is an adjective like 'first', {3} is a day like 'Tuesday', {4} is a time like '12:00pm'",
-        ).t`${"{0}"} ${"{1}"} on the ${"{2}"} ${"{3}"} at ${"{4}"}`,
-        [
-          verb,
-          <SelectFrequency key="frequency" {...frequencyProps} />,
-          <SelectFrame key="frame" {...frameProps} />,
-          <SelectWeekdayOfMonth
-            key="weekday-of-month"
-            {...weekdayOfMonthProps}
-          />,
-          <SelectTime key="time" {...timeProps} />,
-        ],
-      );
-    }
-  } else {
-    return null;
-  }
+  const selectFrame = (
+    <SelectFrame
+      key="frame"
+      schedule={schedule}
+      updateSchedule={updateSchedule}
+    />
+  );
+
+  const selectWeekdayOfMonth = (
+    <SelectWeekdayOfMonth
+      key="wom"
+      schedule={schedule}
+      updateSchedule={updateSchedule}
+    />
+  );
+
+  const { schedule_type, schedule_frame } = schedule;
+
+  return (
+    <Box className={S.Schedule} {...boxProps}>
+      <GroupControlsTogether>
+        {match(schedule_type)
```

Comment:
I propose to wrap this in `useMemo` and move outside of component's markup.

Reply:
Maybe even extract into a custom hook, since it would also require to have all the individual components' variables in there.